internal/pkg/agent/install/uninstall.go

```diff
@@ -35,7 +35,8 @@ import (
 // Uninstall uninstalls persistently Elastic Agent on the system.
 func Uninstall(cfgFile, topPath, uninstallToken string, pt *progressbar.ProgressBar) error {
 	// uninstall the current service
-	svc, err := newService(topPath)
+	// not creating the service, so no need to set the username and group to any value
+	svc, err := newService(topPath, "", "")
```

Comment:
a lot of these with `""` as args while uid and gid are actually used in not that many use cases. maybe we should have `newService(path)` and `newServiceWithOwner(path, uid,gid)` or fluent alternative `newService(path).WithOwner(uid, gid)`

Reply:
I cleaned this up to use the options pattern so it doesn't need to be changed everywhere.